src/index.ts

```diff
@@ -0,0 +1,149 @@
+import escape from 'xml-escape';
+import * as Cypress from 'cypress';
+import RunResult = CypressCommandLine.RunResult;
+import CypressRunResult = CypressCommandLine.CypressRunResult;
+import CypressFailedRunResult = CypressCommandLine.CypressFailedRunResult;
+import BeforeRunDetails = Cypress.BeforeRunDetails;
+import PluginConfigOptions = Cypress.PluginConfigOptions;
+import PluginEvents = Cypress.PluginEvents;
+import Spec = Cypress.Spec;
+import { ConfigOption, JUnitTestSuite } from './type';
+import Reporter, { TestCase, TestSuite } from './reporter';
+
+let reporter: Reporter;
+
+function onBeforeRun(details: BeforeRunDetails) {
+  // The `projectName` is not officially documented in Cypress documentation.
+  // As used in this context, it represents the basename of the current working directory.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  reporter.junitSuite.name = `Cypress Test - ${(details.config as any)?.projectName} ${details.browser?.displayName}`;
+}
+
+function onAfterSpec(spec: Spec, results: RunResult) {
+  const testsuite = new TestSuite();
+  let errCount = 0;
+  results.tests.forEach((test) => {
+    const testcase = new TestCase(
+      // `test.title` includes both the context name (defined in 'describe' or 'context' blocks within Cypress)
+      // and the specific test name.
+      // The last part of `test.title` is the test name itself.
+      // Anything before the last part is the context name.
+      // Consequently, the JUnit testcase name is constructed by combining the context name with the test name.
+      // The JUnit testcase classname is assigned the value of the test name.
```

Comment:
`name` and `classname` is not same thing. This line is for https://github.com/saucelabs/cypress-junit-plugin/pull/1/files/fd1ccb158025daf5eea950401b7f095229ec764d#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R34

Should still be removed?

Reply:
Yeah I'd still remove it. It just explains _what_ it does but not _why_. If we don't need to know _why_ (and the why is arbitrary in this case, isn't it? Since there's no actual classes involved, so we're filling classname with something that feels right enough), then writing the _what_ does not really offer us any benefit.

Reply:
👌